app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt

```diff
@@ -344,6 +345,10 @@ class BrowserFragment : EngineViewLifecycleFragment(), Session.Observer {
                 (cursor?.keyDispatcher?.dispatchKeyEvent(event) ?: false)
     }
 
+    /**
+     * SupportActivity.dispatchKeyEvent() can only be called from within the same library group
+     */
+    @SuppressLint("RestrictedApi")
```

Comment:
@mcomella Yes, you're right. `@SuppressLint("RestrictedApi")` is never a good option. I was initially putting it off since we will be refactoring all of this logic out in re-architecture. I can do more research on better solution though. But this also means we need to change everything related to `fragment?.activity.dispatchKeyEvent`

Reply:
The cheap solution would be to move `dispatchKeyEvent` into `MainActivity` and have everyone else call a wrapper method, I think.

Reply:
Doing a little bit research, it looks like we shouldn't be using generic `activity` type to call its methods from Fragments. Typecasting `MainActivity` works